src/sql/workbench/contrib/query/browser/queryActions.ts

```diff
@@ -323,13 +331,57 @@ export class EstimatedQueryPlanAction extends QueryTaskbarAction {
 		}
 
 		if (this.isConnected(editor)) {
-			editor.input.runQuery(editor.getSelection(), {
-				displayEstimatedQueryPlan: true
-			});
+			editor.input.runQuery(editor.getSelection(), { displayEstimatedQueryPlan: true });
 		}
 	}
 }
 
+/**
+ * Action class that toggles the actual execution plan mode for the editor
+ */
+export class ToggleActualExecutionPlanModeAction extends QueryTaskbarAction {
+	public static EnabledClass = 'enabledActualExecutionPlan';
+	public static ID = 'toggleActualExecutionPlanModeAction';
+
+	private _enableActualPlanLabel = nls.localize('enableActualPlanLabel', "Show Actual Plan");
+	private _disableActualPlanLabel = nls.localize('disableActualPlanLabel', "Hide Actual Plan");
+
+	constructor(
+		editor: QueryEditor,
+		private _isActualPlanMode: boolean,
+		@IQueryManagementService protected readonly queryManagementService: IQueryManagementService,
+		@IConfigurationService protected readonly configurationService: IConfigurationService,
+		@IConnectionManagementService connectionManagementService: IConnectionManagementService,
+		@IAdsTelemetryService private readonly telemetryService: IAdsTelemetryService
+	) {
+		super(connectionManagementService, editor, ToggleActualExecutionPlanModeAction.ID, ToggleActualExecutionPlanModeAction.EnabledClass);
+		this.updateLabel();
+	}
+
+	public get isActualExecutionPlanMode(): boolean {
+		return this._isActualPlanMode;
+	}
+
+	public set isActualExecutionPlanMode(value: boolean) {
+		this._isActualPlanMode = value;
+		this.updateLabel();
+	}
+
+	private updateLabel(): void {
+		// show option to disable actual plan mode if already enabled
+		this.label = this.isActualExecutionPlanMode ? this._disableActualPlanLabel : this._enableActualPlanLabel;
+	}
+
+	public override async run(): Promise<void> {
+		const toActualPlanState = !this.isActualExecutionPlanMode;
+		this.editor.input.state.isActualExecutionPlanMode = toActualPlanState;
+
+		this.telemetryService.createActionEvent(TelemetryKeys.TelemetryView.ExecutionPlan, TelemetryKeys.TelemetryAction.ToggleOnActualExecutionPlan)
```

Comment:
This should be using the `Click` action and then have a target of `ToggleActualExecutionPlan` (remove the On from that - the button is a toggle, not always going to be turning it on)